core/src/main/java/by/ilyin/core/util/error/CustomControllerAdvice.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    //todo add javadoc 401
+
+    @ExceptionHandler(UnauthorizedRequestException.class)
+    public ResponseEntity<ErrorResponse> handleUnauthorizedRequestExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.UNAUTHORIZED);
+    }
+
+    //todo add javadoc 403
+
+    @ExceptionHandler(CustomAccessDeniedException.class)
+    public ResponseEntity<ErrorResponse> handleCustomAccessDeniedExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.FORBIDDEN);
+    }
+
+    //todo add javadoc 404
+
+    @ExceptionHandler(ResourceNotFoundException.class)
+    public ResponseEntity<ErrorResponse> handleResourceNotFoundExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.NOT_FOUND);
+    }
+
+    //todo add javadoc 405
+
+    @ExceptionHandler(HttpRequestMethodNotSupportedException.class)
+    public ResponseEntity<ErrorResponse> handleHttpRequestMethodNotSupportedExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.METHOD_NOT_ALLOWED);
+    }
+
+    //todo add javadoc 406
+
+    @ExceptionHandler(HttpMediaTypeNotAcceptableException.class)
+    public ResponseEntity<ErrorResponse> handleHttpMediaTypeNotAcceptableExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.NOT_ACCEPTABLE);
+    }
+
+    //todo add javadoc 409
+
+    @ExceptionHandler(ResourceAlreadyExists.class)
+    public ResponseEntity<ErrorResponse> handleResourceAlreadyExists(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.CONFLICT);
+    }
+
+    //todo add javadoc 415
+
+    @ExceptionHandler(HttpMediaTypeNotSupportedException.class)
+    public ResponseEntity<ErrorResponse> handleHttpMediaTypeNotSupportedExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.UNSUPPORTED_MEDIA_TYPE);
+    }
+
+    //todo add javadoc 500
+
+    @ExceptionHandler({
+            ConversionNotSupportedException.class,
+            HttpMessageNotWritableException.class,
+            Exception.class
+    })
+    public ResponseEntity<ErrorResponse> handleCInternalServerErrorExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    //todo add javadoc 501
+
+    @ExceptionHandler(NotImplementedMethodException.class)
+    public ResponseEntity<ErrorResponse> handleNotImplementedMethodExceptions(Exception e) {
+        return buildSimpleErrorResponse(e, HttpStatus.NOT_IMPLEMENTED);
+    }
+
+    private ResponseEntity<ErrorResponse> buildSimpleErrorResponse(Exception e, HttpStatus httpStatus) {
+        return buildSimpleErrorResponse(e, httpStatus, e.getMessage());
+    }
+
+    private ResponseEntity<ErrorResponse> buildSimpleErrorResponse(Exception e, HttpStatus httpStatus, String message) {
+        return new ResponseEntity<>(
+                new ErrorResponse(
+                        e.getStackTrace(),
+                        httpStatus,
```

Comment:
@YauhenIlyin For what reason you have http status in error response DTO?

Reply:
@DzmitryKazakevich it's specified in the task 
![image](https://user-images.githubusercontent.com/114590493/207131683-8ba34dfc-24b8-4c3d-bb5d-0e7346e34341.png)